Chapter 06 examples

villares opened the pull request.

villares committed:
Merge pull request #1 from shiffman/master

atualizando

villares committed:
chp07_CA GOL

villares committed:
Merge pull request #2 from shiffman/master

update from Shiffman

villares committed:
Delete .DS_Store

villares committed:
7_01 Wolfram

villares committed:
jdf finished CA scroll

villares committed:
Update CA.py

villares committed:
Update CA.py

villares committed:
Update CA.py

villares committed:
Update CA.py

villares committed:
seek example

villares committed:
Arrive

villares committed:
stay within walls

jdf commented on chp06_agents/NOC_06_01_Seek/NOC_06_01_Seek.pyde:
```diff
@@ -0,0 +1,33 @@
+
+# Implements Craig Reynold's autonomous steering behaviors
+# One vehicle "seeks"
+# See: http:#www.red3d.com/cwr/
```
//

villares replied:
fixed!

jdf commented on chp06_agents/NOC_06_01_Seek/NOC_06_01_Seek.pyde:
```diff
@@ -0,0 +1,33 @@
+# One vehicle "seeks"
+# See: http:#www.red3d.com/cwr/
+
+from Vehicle import Vehicle
```
In general, python modules (and therefore the implementation file names) begin with a lower-case letter.

villares replied:
I think I have fixed this, I'm not sure (branching nightmare)

jdf commented on chp06_agents/NOC_06_01_Seek/Vehicle.py:
```diff
@@ -0,0 +1,59 @@
+# The "Vehicle" class
+
+class Vehicle():
+    def __init__(self,x, y):
```
Auto-format this and all files.

villares replied:
Sorry, will do that.

jdf commented on chp06_agents/NOC_06_01_Seek/Vehicle.py:
```diff
@@ -0,0 +1,59 @@
+
+        # Steering = Desired minus velocity
+        steer = desired - self.velocity
+        steer.limit(self.maxforce)  # Limit to maximum steering force
```
This comment doesn't add anything.

jdf reviewed (commented):
Thanks for doing this!

villares committed:
fixing small things

3 -> // on URLs
lowercase module names
fixed typo on a method name
removed steering comment

villares committed:
auto-format

jdf commented:
I see you're making chunks of changes; just send me a comment when you're ready for further review.

villares commented:
Thanks, I think those committed are OK now, next week I might be able to add more examples.

jdf commented on chp07_CA/Exercise_7_01_WolframCA_randomizedrules/CA.py:
```diff
@@ -0,0 +1,92 @@
+# Daniel Shiffman
+# http://natureofcode.com
+
+# Wolfram Cellular Automata
```
I thought this pull request was about chapter 6 examples. Aren't these chapter 7 examples in another pull request?

villares commented:
This is probably my git incompetence showing, sorry. (Confused by branches).

On Sun, 1 Jul 2018, 23:42 Jonathan Feinberg, <notifications@github.com>
wrote:

> *@jdf* commented on this pull request.
> ------------------------------
>
> In chp07_CA/Exercise_7_01_WolframCA_randomizedrules/CA.py
> <https://github.com/shiffman/The-Nature-of-Code-Examples-Python/pull/7#discussion_r199374472>
> :
>
> > @@ -0,0 +1,92 @@
> +# The Nature of Code
> +# Daniel Shiffman
> +# http://natureofcode.com
> +
> +# Wolfram Cellular Automata
>
> I thought this pull request was about chapter 6 examples. Aren't these
> chapter 7 examples in another pull request?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/shiffman/The-Nature-of-Code-Examples-Python/pull/7#pullrequestreview-133480634>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADhgDPsDkfPOZDlEgpPxxOewyjMt24jPks5uCYiBgaJpZM4U23KS>
> .
>

villares committed:
constant name change & import style

w -> cell_size
Changed import *

villares committed:
Rename Cell.py to cell.py

villares committed:
Update GOL.py

villares commented:
Look, I have no idea how to work on branches and stuff... I have tried to fix/remove the `import *` issue again. Please bail me out so I can start fresh.

villares commented:
Means help me sort the multiple branch/pull request nightmare.

On 8 August 2018 at 11:19, Jonathan Feinberg <notifications@github.com>
wrote:

> I don't know what you mean by "bail you out".
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/shiffman/The-Nature-of-Code-Examples-Python/pull/7#issuecomment-411422507>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADhgDNL6HIjtWl1yilbYrYtVkhU4418wks5uOvN8gaJpZM4U23KS>
> .
>

villares committed:
Update CA.py

villares committed:
Merge branch 'master' into ch_06

villares committed:
Rename Vehicle.py to vehicle.py
